### PR TITLE
Storages: Refine SegmentReadTaskPool and SegmentReadTaskScheduler

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #6834

### What is changed and how it works?

- This PR mainly refines `SegmentReadTaskPool` and `SegmentReadTaskScheduler`  preparing for supporting read-thread and data-sharing in disaggregated-mode. The logic of the reading path is still **not changed** after this PR.

Main changes:
- Remove `DMContext` and `physical_table_id` from `SegmentReadTaskPool`.
- Add DMContext to `SegmentReadTask`.
  - In disaggregated-mode, `EstablishDisaggTask` can return `SegmentReadTasks` of different physical tables. So `SegmentReadTaskPool` needs to support `SegmentReadTasks` of different physical tables.
  - Move information of physical tables in `DMContext` into `SegmentReadTask` .
- Use `GlobalSegmentID` to identify a segment in `SegmentReadTaskScheduler`.
  - In op-mode, `physical_table_id` + `segment_id` + `segment_epoch` can identify a segment. In disaggregated-mode, it also needs to add `store_id` and `keyspace_id`.
  - Wrap this information into `struct GlobalSegmentID`.
- Enhance the formatter of `SegmentReadTaskPtr` and remove `SegmentReadTask::info`.
- Remove `cluster` member from `RNWorkerFetchPages`.
- Remove `RNReadTask` since it is redundant.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/SegmentReadTask.h [hunk 1]
```diff
@@ -113,21 +102,84 @@ struct SegmentReadTask
...
+    static constexpr auto parse(format_parse_context & ctx) { return ctx.begin(); }
+
     template <typename FormatContext>
-    auto format(const DB::DM::SegmentReadTaskPtr & t, FormatContext & ctx) const -> decltype(ctx.out())
+    auto format(const DB::DM::SegmentReadTaskPtr & t, FormatContext & ctx) const
     {
         return format_to(
             ctx.out(),
-            "{}_{}_{}",
+            "s{}_k{}_t{}_{}_{}_{}",
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SegmentReadTask.h:133]:
```suggestion
            "s{}_ks{}_t{}_{}_{}_{}",
```
align with the naming in the physical directory name

File: dbms/src/Storages/DeltaMerge/SegmentReadTask.h [hunk 2]
```diff
@@ -113,21 +102,84 @@ struct SegmentReadTask
...
+struct fmt::formatter<DB::DM::GlobalSegmentID>
+{
+    static constexpr auto parse(format_parse_context & ctx) { return ctx.begin(); }
+
+    template <typename FormatContext>
+    auto format(const DB::DM::GlobalSegmentID & t, FormatContext & ctx) const
+    {
+        return format_to(
+            ctx.out(),
+            "s{}_k{}_t{}_{}_{}",
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SegmentReadTask.h:153]:
```suggestion
            "s{}_ks{}_t{}_{}_{}",
```

File: dbms/src/Storages/DeltaMerge/SegmentReadTask.h [hunk 3]
```diff
@@ -27,40 +27,33 @@ struct SegmentReadTask;
...
-    String toString() const
-    {
-        return fmt::format(
-            "RemoteSegmentID<store_id={} keyspace={} table_id={} segment_id={}>",
-            store_id,
-            keyspace_id,
-            physical_table_id,
-            segment_id);
-    }
+// A SegmentReadTask object is identified by <store_id, keyspace_id, physical_table_id, segment_id, segment_epoch>.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SegmentReadTask.h:31]:
```suggestion
// A SegmentReadTask object is identified by <store_id, keyspace_id, physical_table_id, segment_id, segment_epoch>.
// Under disagg arch, there could be SegmentReadTasks from different stores in one compute node.
```

File: dbms/src/Storages/DeltaMerge/ReadThread/SegmentReader.cpp [hunk 4]
```diff
@@ -222,7 +222,7 @@ void SegmentReaderPoolManager::init(UInt32 logical_cpu_cores, double read_thread
 
 void SegmentReaderPoolManager::addTask(MergedTaskPtr && task)
 {
-    static std::hash<uint64_t> hash_func;
+    static std::hash<GlobalSegmentID> hash_func;
     auto idx = hash_func(task->getSegmentId()) % reader_pools.size();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ReadThread/SegmentReader.cpp:226]:
```suggestion
    auto idx = std::hash<GlobalSegmentID>(task->getSegmentId()) % reader_pools.size();
```

**Review by JaySon-Huang — COMMENTED** [review]
Rest LGTM

File: dbms/src/Storages/DeltaMerge/Remote/RNWorkerFetchPages.cpp [hunk 5]
```diff
@@ -188,7 +187,7 @@ void RNWorkerFetchPages::doFetchPages(
     double total_write_page_cache_sec = 0.0;
 
     pingcap::kv::RpcCall<pingcap::kv::RPC_NAME(FetchDisaggPages)> rpc(
-        cluster->rpc_client,
+        seg_task->dm_context->db_context.getTMTContext().getKVCluster()->rpc_client,
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/Remote/RNWorkerFetchPages.cpp:190]: OK

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**